### PR TITLE
Support added for Python3.10's Union Pipe operator

### PR DESCRIPTION
**Pull request by ZhangZhida**
*Issue number of the reported bug or feature request: #80*

**Describe your changes**
This PR is based off the original PR https://github.com/bloomberg/attrs-strict/pull/102 because I have permission problem updating that PR.

In order to resolve the issue - Support of pipe symbol for Union operation. Python3.10 uses base type as types.UnionTypes. So therefore added a case for that (If the base type is types.UnionTypes then perform validation). Hence supporting Union as X | Y.

**Testing performed**
Testing performed on Mac OS 13.3.1 and Python 3.10.7

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

**Review by gaborbernat — APPROVED** [review]

File: src/attrs_strict/_type_validation.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 from itertools import zip_longest
 from typing import ForwardRef
 
+if sys.version_info >= (3, 10):
```
Comment (gaborbernat) [inline src/attrs_strict/_type_validation.py:13]: `# pragma: >=3.10 cover`

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/attrs_strict/_type_validation.py (hunk 1 above)
Comment (ZhangZhida) [reply src/attrs_strict/_type_validation.py]: Added

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**